.ci/install_kata_image.sh

```diff
@@ -95,6 +97,48 @@ check_not_empty() {
 	fi
 }
 
+build_image_with_dracut() {
+	image_output=${1}
+	os_version=${2}
+	agent_commit=${3}
+
+	check_not_empty "$image_output" "Missing image"
+	check_not_empty "$os_version" "Missing os version"
+	check_not_empty "$agent_commit" "Missing agent commit"
+
+	pushd "${osbuilder_path}" >/dev/null
+
+	if [ "${TEST_INITRD}" == "yes" ]; then
+		local local image_name="kata-containers-initrd.img"
+		local make_target="initrd"
+	else
+		local make_target="image"
+		local image_name="kata-containers.img"
+	fi
+
+	sudo -E AGENT_VERSION="${agent_commit}" PATH="$PATH" GOPATH="$GOPATH" OS_VERSION=${os_version} make BUILD_METHOD=dracut AGENT_INIT="${AGENT_INIT}" "$make_target"
+
+	sudo install -o root -g root -m 0640 -D ${image_name} "${IMAGE_DIR}/${image_output}"
+
+	# Unlike the distro method, the dracut-based method doesn't leave
+	# behind an unpacked root filesystem tree, so to get access to
+	# osbuilder.yaml in order to install it, we need to extract it to
+	# a temporary directory from the initrd.
+	tmpdir=$(mktemp -d)
+	local image_abs_name=${PWD}/${image_name}
+	pushd ${tmpdir} >/dev/null
+	cat ${image_abs_name} | cpio -idmv var/lib/osbuilder/osbuilder.yaml
+	popd >/dev/null
+
+	sudo install -o root -g root -m 0640 -D "${tmpdir}/var/lib/osbuilder/osbuilder.yaml" "${IMAGE_DIR}/${OSBUILDER_YAML_INSTALL_NAME}"
+
+	rm -rf ${tmpdir}
```

Comment:
Won't this fail given that the build was done as root above? I agree with @c3d that the build should be run as a non-priv user if at all possible.

Reply:
@pmores, I guess this is the last missing bit that has to be addressed.

Reply:
Hmm. Seems that just a reply was missing. It does make sense and seems good to go.

Reply:
@jodh-intel, I've marked this conversation (and the others) as resolved.
Please, feel free to "unresolve" them.